src/tx-builder.ts

```diff
@@ -0,0 +1,314 @@
+import { execSync } from 'child_process';
+import path from 'path';
+import { bcs, BcsType } from '@mysten/bcs';
+import { SuiClient, SuiMoveNormalizedType, SuiObjectChange, SuiTransactionBlockResponseOptions } from '@mysten/sui.js/client';
+import { Keypair } from '@mysten/sui.js/dist/cjs/cryptography';
+import { TransactionBlock, TransactionObjectInput } from '@mysten/sui.js/transactions';
+import { utils as ethersUtils } from 'ethers';
+import tmp from 'tmp';
+import { updateMoveToml } from './utils';
+
+const { arrayify, hexlify } = ethersUtils;
+
+const objectCache = {} as { [id in string]: SuiObjectChange };
+
+function updateCache(objectChanges: SuiObjectChange[]) {
+    for (const change of objectChanges) {
+        if (
+            !(
+                change as {
+                    objectId: string;
+                }
+            ).objectId
+        )
+            continue;
+        objectCache[
+            (
+                change as {
+                    objectId: string;
+                }
+            ).objectId
+        ] = change;
+    }
+}
+
+function getObject(tx: TransactionBlock, object: TransactionObjectInput) {
```

Comment:
could you please explicitly declare the return type here an other functions as well?